scipy/optimize/cython_optimize/examples/zeros_array_examples.pyx

```diff
@@ -0,0 +1,143 @@
+"""
```

Comment:
Solving the single diode equation seems unnecessarily complex for an intro example. It's hard for me to follow despite being familiar with the application. I think this could be a good "real world" example if presented with more context.

Reply:
Could be helpful to mirror some examples from https://www.scipy-lectures.org/advanced/mathematical_optimization/

Reply:
Maybe only a single optimizer function in the simplest possible example (choose only one of newton, brent, etc).

Reply:
I'm ambivalent. I agree with you on needs more context and/or too complex, but the examples also serve the purpose of testing all 21 combinations of 7 root-finders & 3 callback signatures so I'm not in favor of removing them. I was thinking of adding more context either in the `examples/__init__.py` or in a tutorial page similar to `cython_special`. A few others, `ndimage` and `integrate.quad` also have good examples, but all of them assume the reader can write and compile a cython package

Reply:
I think you should keep the examples you've already made. What I'd find useful is a single file that tests a single root finder with a single callback signature with a single simple, generic function. Brent and either array or struct seems most useful. I'll check out the other subpackages though!

Reply:
I added simple examples to the [Jupyter notebook in the examples folder](../../../mikofski/scipy/blob/cython_optimize/scipy/optimize/cython_optimize/examples/cython_optimize.ipynb). Hope that's better?